llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -22142,6 +22142,10 @@ static SDValue performVectorShiftCombine(SDNode *N,
     if (DCI.DAG.ComputeNumSignBits(Op.getOperand(0)) > ShiftImm)
       return Op.getOperand(0);
 
+  // If the right shift is exact, the shifted out bits matter.
```

Comment:
(Somehow didn't send this out before -- posting for posterity.)

Reply:
I wrote that in the commit message

> However, these elements need to stay zero if the right shift is exact, or otherwise we will be introducing undefined behavior.